deepchecks/checks/methodology/model_inference_time.py

```diff
@@ -0,0 +1,94 @@
+"""The model inference time check module."""
+import typing as t
+import timeit
+
+import pandas as pd
+
+from deepchecks import SingleDatasetBaseCheck, CheckResult, Dataset, ConditionResult
+from deepchecks.utils import model_type_validation
+from deepchecks.string_utils import format_number
+
+
+__all__ = ['ModelInferenceTimeCheck']
+
+
+MI = t.TypeVar('MI', bound='ModelInferenceTimeCheck')
+
+
+class ModelInferenceTimeCheck(SingleDatasetBaseCheck):
+    """Measure model average inference time (in seconds) per sample.
+
+    Args:
+        number_of_samples (int): number of samples to use for inference
+    """
+
+    def __init__(self, number_of_samples: int = 1000):
+        self.number_of_samples = number_of_samples
+        super().__init__()
+
+    def run(self, dataset: Dataset, model: object) -> CheckResult:
+        """Run check.
+
+        Args:
+            dataset (Dataset): samples that will be used to measure inference time
+            model (BaseEstimator): a model to measure inference time
+
+        Returns:
+            CheckResult:
+                value is of the type 'float'
+
+        Raises:
+            DeepchecksValueError: If the 'test_dataset' is not a 'Dataset' instance with a label or
+                if 'model' is not a scikit-learn-compatible fitted estimator instance
+        """
+        return self._model_inference_time_check(dataset, model)
```

Comment:
I'm not a fan of this split, is there a reason for it or just convention from other checks?

Reply:
ust convention from other checks

Reply:
not all checks do it I think. I think it should be reserved to cases when there is actual separation of function